relay/config/dynamic_config.go

```diff
@@ -7,47 +7,106 @@ import (
 	"os"
 	"path"
 	"strings"
+	"fmt"
 )
 
 // LoadDynamicConfig loads the dyanmic configuration for a bundle if
 // a) dynamic configuration is enabled and b) a config file exists for
-// the requested bundle
-func (c *Config) LoadDynamicConfig(bundle string) map[string]interface{} {
+// the requested bundle. Room- and user-specific configurations are
+// layered on top, if they exist.
+//
+// If any configuration files exist, but cannot be properly processed
+// (read, parsed as YAML, etc), an empty map is returned.
+func (c *Config) LoadDynamicConfig(bundle string, roomName string, userName string) map[string]interface{} {
 	retval := make(map[string]interface{})
 	if c.DynamicConfigRoot == "" {
 		log.Debugf("Dynamic config disabled.")
 		return retval
 	}
-	if fullPath := locateConfigFile(c.DynamicConfigRoot, bundle); fullPath != "" {
-		buf, err := ioutil.ReadFile(fullPath)
+
+	rootInfo, rootErr := os.Stat(c.DynamicConfigRoot)
+	if rootErr != nil || rootInfo.IsDir() == false {
+		log.Debugf("Dynamic configuration root dir not found.")
+		return retval
+	}
+
+	configs := allConfigFiles(c.DynamicConfigRoot, bundle, roomName, userName)
+	for _, f := range configs {
+		currentYAML := make(map[string]interface{})
+
```

Comment:
We should also log which files are being read in addition to the error logging. `debug` log level seems appropriate.